framework/mesh/mesh_generator/distributed_mesh_generator.cc

```diff
@@ -0,0 +1,299 @@
+    serial_data.Write<unsigned int>(umesh.Dimension());
+    serial_data.Write(static_cast<int>(umesh.Type()));
+    serial_data.Write(umesh.Extruded());
+    auto& ortho_attrs = umesh.OrthoAttributes();
+    serial_data.Write(ortho_attrs.Nx);
+    serial_data.Write(ortho_attrs.Ny);
+    serial_data.Write(ortho_attrs.Nz);
+    serial_data.Write(raw_vertices.size());
+
+    // Boundaries
+    const auto& bndry_map = umesh.BoundaryIDMap();
+    serial_data.Write(bndry_map.size());
+    for (const auto& [bid, bname] : bndry_map)
+    {
+      serial_data.Write(bid);
+      const size_t num_chars = bname.size();
+      serial_data.Write(num_chars);
+      for (size_t i = 0; i < num_chars; ++i)
+        serial_data.Write(bname.data()[i]);
+    }
+
+    // Number of cells and vertices
+    serial_data.Write(cells_needed.size());
+    serial_data.Write(vertices_needed.size());
+
+    // Cell data
+    for (const auto& cell_global_id : cells_needed)
+    {
+      const auto& cell = *raw_cells[cell_global_id];
+      serial_data.Write(static_cast<int>(cell_pids[cell_global_id]));
+      serial_data.Write(cell_global_id);
+      serial_data.Write(cell.type);
+      serial_data.Write(cell.sub_type);
+      serial_data.Write(cell.centroid.x);
+      serial_data.Write(cell.centroid.y);
+      serial_data.Write(cell.centroid.z);
+      serial_data.Write(cell.material_id);
+      serial_data.Write(cell.vertex_ids.size());
+      for (uint64_t vid : cell.vertex_ids)
+        serial_data.Write(vid);
+
+      serial_data.Write(cell.faces.size());
+      for (const auto& face : cell.faces)
+      {
+        serial_data.Write(face.vertex_ids.size());
+        for (uint64_t vid : face.vertex_ids)
+          serial_data.Write(vid);
+        serial_data.Write(face.has_neighbor);
+        serial_data.Write(face.neighbor);
+      }
+    }
+
+    // Vertex data
+    for (uint64_t vid : vertices_needed)
+    {
+      serial_data.Write(vid);
+      serial_data.Write(raw_vertices[vid]);
+    }
+
+    if (pid == 0)
+      loc0_data = serial_data;
+    else
+      opensn::mpi_comm.send<std::byte>(pid, pid, serial_data.Data().data(), serial_data.Size());
+  }
+
+  return loc0_data;
+}
+
+DistributedMeshGenerator::DistributedMeshData
+DistributedMeshGenerator::DeserializeMeshData(ByteArray& serial_data)
+{
+  DistributedMeshData info_block;
+
+  // Basic mesh data
+  info_block.dimension = serial_data.Read<unsigned int>();
+  info_block.mesh_type = static_cast<MeshType>(serial_data.Read<int>());
+  info_block.extruded = serial_data.Read<bool>();
+  info_block.ortho_attributes.Nx = serial_data.Read<size_t>();
+  info_block.ortho_attributes.Ny = serial_data.Read<size_t>();
+  info_block.ortho_attributes.Nz = serial_data.Read<size_t>();
+  info_block.num_global_vertices = serial_data.Read<size_t>();
+
+  // Boundaries
+  const size_t num_bndries = serial_data.Read<size_t>();
+  for (size_t b = 0; b < num_bndries; ++b)
+  {
+    const uint64_t bid = serial_data.Read<uint64_t>();
```

Comment:
On calls like this, you can safely use `auto`, since the compiler figures out the type from the template parameter (I think there is a clang-tidy switch for things like this). I suggest to use `auto` in all of these calls..

Reply:
Fixed.

Reply:
This fix also needs to happen on lines: 218, 219, 224, 225, 240, 244, 259.

Reply:
Fixed.